File: backend/src/models/inspection_task.py
from pydantic import BaseModel


class InspectionTask(BaseModel):
    id: int
    building_id: int
    inspector_id: int
    plan_date: str
    task_type: str
    status: str                  # PLANNED / IN_PROGRESS / SUBMITTED / REVIEWED / OVERDUE
    checklist_version: str
    finished_at: str
    device_ids: list[int] = []

File: frontend/src/types/InspectionTask.ts
export interface InspectionTask {
  id: number;
  building_id: number;
  inspector_id: number;
  plan_date: string;
  task_type: string;
  status: string;              // PLANNED / IN_PROGRESS / SUBMITTED / REVIEWED / OVERDUE
  checklist_version: string;
  finished_at: string;
  device_ids: number[];
}
